### DISCUSSION TITLE
How should I configure my barrier with dynamic ip on server side?

### DISCUSSION THREAD

**Discussion by decoherencer** [debauchee/barrier#1607] [Q&A]
I have a laptop that I use as a barrier server, but its IP gets updated every few hours and the client, which has static IP, is on the same local network but with different subnets. 
I tried configuring a cron job that updates a DynDNS service every x minutes which is then used as a client-side URL for IP, 
and I also tried setting up a hook on my arch when the wifi restarts from the NetworkManager dispatcher
but I would like a smooth transition, such that my server automatically stays connected with the client, with IP configured automatically i.e., only whenever server IP changes this info gets relayed to the client properly and ideally once. Any suggestions?
I do want to use only my laptop as a server for keyboard capabilities and I can't create static IPs for laptop

Thank you

**Comment by skipmorris** [comment]
I use ssh tunnels to solve that problem.  The machine with the moving IP does an ssh connection to the  other machine (which has inbound ssh enabled); ssh keys are used to validate access.  the tunnel is established using the command:
ssh -f -A -R 24801:localhost:24800 ${REMOTEUSERNAME}@${STATICIPADDRESS} -p $REMOTEINBOUNDSSHPORT "-N"

Then Barrier at each end just connects to "localhost:24800" or "localhost:24801".
